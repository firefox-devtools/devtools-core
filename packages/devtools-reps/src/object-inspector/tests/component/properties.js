/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

 /* global jest */

const { mount } = require("enzyme");
const React = require("react");
const { createFactory } = React;
const ObjectInspector = createFactory(require("../../index"));

const gripRepStubs = require("../../../reps/stubs/grip");

describe("ObjectInspector - properties", () => {
  it("does not load properties if getObjectProperties returns a truthy element", () => {
    const stub = gripRepStubs.get("testMaxProps");
    const loadObjectProperties = jest.fn();

    mount(ObjectInspector({
      autoExpandDepth: 1,
      roots: [{
        path: "root",
        contents: {
          value: stub
        }
      }],
      getObjectProperties: actor => {
        return {
          ownProperties: stub.preview.ownProperties,
        };
      },
      loadObjectProperties,
    }));

    expect(loadObjectProperties.mock.calls.length).toBe(0);
  });

  it("calls loadObjectProperties when expandable leaf is clicked", () => {
    const stub = gripRepStubs.get("testMaxProps");
    const loadObjectProperties = jest.fn();

    const oi = mount(ObjectInspector({
      autoExpandDepth: 0,
      roots: [{
        path: "root",
        contents: {
          value: stub
        }
      }],
      getObjectProperties: () => {},
      loadObjectProperties,
    }));

    const node = oi.find(".node");
    node.simulate("click");

    expect(loadObjectProperties.mock.calls.length).toBe(1);
  });
});
