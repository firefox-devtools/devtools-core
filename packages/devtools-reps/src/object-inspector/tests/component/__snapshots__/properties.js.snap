// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector - properties renders uninitialized bindings 1`] = `"<div class=\\"tree object-inspector\\" role=\\"tree\\" tabindex=\\"0\\" style=\\"padding: 0px; margin: 0px;\\"><div id=\\"root/someFoo\\" class=\\"tree-node\\" role=\\"treeitem\\" aria-level=\\"0\\" data-expandable=\\"false\\"><div class=\\"node object-node\\"><span class=\\"object-label\\">someFoo</span><span class=\\"object-delimiter\\">: </span><span class=\\"unavailable\\">(uninitialized)</span></div></div></div>"`;

exports[`ObjectInspector - properties renders unmapped bindings 1`] = `"<div class=\\"tree object-inspector\\" role=\\"tree\\" tabindex=\\"0\\" style=\\"padding: 0px; margin: 0px;\\"><div id=\\"root/someFoo\\" class=\\"tree-node\\" role=\\"treeitem\\" aria-level=\\"0\\" data-expandable=\\"false\\"><div class=\\"node object-node\\"><span class=\\"object-label\\">someFoo</span><span class=\\"object-delimiter\\">: </span><span class=\\"unavailable\\">(unmapped)</span></div></div></div>"`;

exports[`ObjectInspector - properties renders unscoped bindings 1`] = `"<div class=\\"tree object-inspector\\" role=\\"tree\\" tabindex=\\"0\\" style=\\"padding: 0px; margin: 0px;\\"><div id=\\"root/someFoo\\" class=\\"tree-node\\" role=\\"treeitem\\" aria-level=\\"0\\" data-expandable=\\"false\\"><div class=\\"node object-node\\"><span class=\\"object-label\\">someFoo</span><span class=\\"object-delimiter\\">: </span><span class=\\"unavailable\\">(unscoped)</span></div></div></div>"`;
